src/unix/timer.c

```diff
@@ -71,7 +72,7 @@ int uv_timer_start(uv_timer_t* handle,
   if (uv__is_active(handle))
     uv_timer_stop(handle);
 
-  clamped_timeout = handle->loop->time + timeout;
+  clamped_timeout = (handle->timeout <= 0 ? handle->loop->time : handle->timeout) + timeout;
```

Comment:
This PR is aimed to fix next loops of repeating timers, not the first loop.
for example:
I need a repeating timer which runs every exact minute, so I start a timer with uv_timer_start with timout 60*1000 and repeat 60*1000. While timer is triggered, my callback function is executed for about 3 seconds, then the next loop of timer would be delayed by 3 seconds, coz the original codes add interval to the time on the callback ends.